exchanges/orderbook/depth_test.go

```diff
@@ -0,0 +1,331 @@
+package orderbook
+
+import (
+	"errors"
+	"log"
+	"reflect"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/gofrs/uuid"
+	"github.com/thrasher-corp/gocryptotrader/currency"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/asset"
+)
+
+var id, _ = uuid.NewV4()
+
+func TestGetLength(t *testing.T) {
+	d := newDepth(id)
+	if d.GetAskLength() != 0 {
+		t.Errorf("expected len %v, but received %v", 0, d.GetAskLength())
+	}
+
+	d.asks.load([]Item{{Price: 1337}}, d.stack)
+
+	if d.GetAskLength() != 1 {
+		t.Errorf("expected len %v, but received %v", 1, d.GetAskLength())
+	}
+
+	d = newDepth(id)
+	if d.GetBidLength() != 0 {
+		t.Errorf("expected len %v, but received %v", 0, d.GetBidLength())
+	}
+
+	d.bids.load([]Item{{Price: 1337}}, d.stack)
+
+	if d.GetBidLength() != 1 {
+		t.Errorf("expected len %v, but received %v", 1, d.GetBidLength())
+	}
+}
+
+func TestRetrieve(t *testing.T) {
+	d := newDepth(id)
+	d.asks.load([]Item{{Price: 1337}}, d.stack)
+	d.bids.load([]Item{{Price: 1337}}, d.stack)
+	d.options = options{
+		exchange:              "THE BIG ONE!!!!!!",
+		pair:                  currency.NewPair(currency.THETA, currency.USD),
+		asset:                 "Silly asset",
+		lastUpdated:           time.Now(),
+		lastUpdateID:          007,
+		notAggregated:         true,
+		isFundingRate:         true,
+		verificationBypass:    true,
+		hasChecksumValidation: true,
+		restSnapshot:          true,
+		idAligned:             true,
+	}
+
+	// If we add anymore options to the options struct later this will complain
+	// generally want to return a full carbon copy
+	mirrored := reflect.Indirect(reflect.ValueOf(d.options))
+	for n := 0; n < mirrored.NumField(); n++ {
+		structVal := mirrored.Field(n)
+		if structVal.IsZero() {
+			t.Fatalf("struct value options not set for field %v",
+				mirrored.Type().Field(n).Name)
+		}
+	}
```

Comment:
Could you clarify this? reflect indirect checks a pointer's value, if its 0 it has no value, but why would adding more fields affect this? d.options also isn't a pointer Is this test just a barrier test to prevent someone from adding too many fields to the struct?

Reply:
No it was just for me really when I was adding in new items options so I remembered to add the assignments over as well, I can remove it if you'd like as its not needed: 
If I add in field gloriousField to the options struct and then run the tests it will complain:
![image](https://user-images.githubusercontent.com/9391715/111261594-d0aa0b80-8676-11eb-9001-7f35965bf08f.png)

Reply:
That's cool. It can always become its own dedicated test